src/main/java/edu/iis/powp/adapter/DrawerToMagicPlotterAdapter.java

```diff
@@ -30,14 +31,15 @@ public void drawTo(int x, int y)
         ILine line = LineFactory.getBasicLine();
     	line.setStartCoordinates(this.startX, this.startY);
         line.setEndCoordinates(x, y);
+
         setPosition(x,y);
 
-		drawLine(line);
+        ApplicationWithDrawer.getDrawPanelController().drawLine(line);
```

Comment:
1. Constructor injection should be used (using a parameter). This is unnecessary direct coupling with ApplicationWithDrawer class.
2. Inheritance is not used (no super drawLine calls). Should be removed!